lib/content-services/src/lib/category/categories-management/categories-management.component.ts

```diff
@@ -0,0 +1,287 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Category } from '@alfresco/js-api';
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output, ViewEncapsulation } from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { MatSelectionListChange } from '@angular/material/list';
+import { EMPTY, Observable, Subject, timer } from 'rxjs';
+import { debounce, first, map, takeUntil, tap } from 'rxjs/operators';
+import { CategoriesManagementMode } from '../categories-management-mode';
+import { CategoryService } from '../services/category.service';
+
+interface CategoryNameControlErrors {
+    duplicatedExistingCategory?: boolean;
+    duplicatedCategory?: boolean;
+    emptyCategory?: boolean;
+    required?: boolean;
+}
+
+@Component({
+  selector: 'adf-categories-management',
+  templateUrl: './categories-management.component.html',
+  styleUrls: ['./categories-management.component.scss'],
+  encapsulation: ViewEncapsulation.None
+})
+export class CategoriesManagementComponent implements OnInit, OnDestroy {
+    readonly nameErrorMessagesByErrors = new Map<keyof CategoryNameControlErrors, string>([
+        ['duplicatedExistingCategory', 'ALREADY_EXISTS'],
+        ['duplicatedCategory', 'DUPLICATED_CATEGORY'],
+        ['emptyCategory', 'EMPTY_CATEGORY'],
+        ['required', 'REQUIRED']
+    ]);
+
+    private existingCategoryLoaded$ = new Subject<void>();
+    private cancelExistingCategoriesLoading$ = new Subject<void>();
+    private onDestroy$ = new Subject<void>();
+    private _categoryNameControl = new FormControl<string>(
+        '',
+        [
+            this.validateIfNotAlreadyAdded.bind(this),
+            this.validateEmptyCategory
+        ],
+        this.validateIfNotAlreadyCreated.bind(this)
+    );
+    private _existingCategories: Category[];
+    private _categoryNameErrorMessageKey = '';
+    private _existingCategoriesLoading = false;
+    private _typing = false;
+    private _existingCategoriesPanelVisible: boolean;
+    existingCategoriesListLimit = 15;
+    noCategoriesTitle = '';
+    removeCategoryTitle = '';
+    initialCategories: Category[] = [];
+
+    @Input()
+    categories: Category[] = [];
```

Comment:
Could you add jsdoc to whole public API? It was commented by Denys too here https://github.com/Alfresco/alfresco-ng2-components/pull/8417#discussion_r1151149349. 

So please add doc to inputs, outputs, public functions etc.

Reply:
Added